Fixed admin product image placeholders scope.

ivanmiskic opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->
Added "storeId" attribute to scope config "getValue" method to pull the scope related placeholders for admin product images in product listing grid.

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. none, reason for change is that magento is showing product placeholders in admin without taking into consideration store view request filter. Ending up with wrong placeholder values from different stores.

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. Create two store views
2. Upload different product placeholder images
3. Create one product per store view
4. Filter products by store view
5. Observe correct placeholders are shown with my fix.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

ivanmiskic committed:
Fixed admin product image placeholders scope.

Added "storeId" attribute to scope config "getValue" method to pull the scope related placeholders for admin product images in product listing grid.

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=21677) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=21677) before we can accept your contribution.<br/>

magento-engcom-team commented:
Hi @ivanmiskic. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

ivanmiskic commented:
@magento-engcom-team give me test instance

magento-engcom-team commented:
Hi @ivanmiskic. Thank you for your request. I'm working on Magento instance for you

magento-engcom-team commented:
Hi @ivanmiskic, here is your [new Magento instance](https://pr-21677.instances.magento-community.engineering).
Admin access: [https://pr-21677.instances.magento-community.engineering/admin](https://pr-21677.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

ivanmiskic committed:
Updating Test file

ivanmiskic committed:
Fix

ivanmiskic committed:
Update PlaceholderTest.php

dmytro-ch commented on app/code/Magento/Catalog/Model/View/Asset/Placeholder.php:
```diff
@@ -53,24 +54,32 @@ class Placeholder implements LocalInterface
         ContextInterface $context,
         ScopeConfigInterface $scopeConfig,
         Repository $assetRepo,
+        Http $request,
```
The new constructor parameters should be optional in order to avoid backward compatibility issues.
Please check the [Contribution Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/#php) (section *Adding a constructor parameter*) for more details.

ivanmiskic replied:
I believe you are referring to the order of the arguments, simply placing it as last would make it optional in this case?

dmytro-ch replied:
Yes, but not exactly.
The idea is to fetch the dependency using `Magento\Framework\App\ObjectManager::getInstance()`, if the new dependency is not provided (i.e. the value of the introduced argument is null).
Please check the example code [here](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/#adding-a-constructor-parameter).

dmytro-ch commented on app/code/Magento/Catalog/Model/View/Asset/Placeholder.php:
```diff
@@ -152,10 +161,13 @@ public function getFilePath()
             return $this->filePath;
         }
+
+        $storeId = $this->request->getParam('filters')['store_id'];
```
I guess we need to make sure the value of `filters` parameter is an array and `store_id` key exists in order to avoid potential exceptions.

ivanmiskic replied:
I attempted to do this, let me know what you think.
Thanks.

dmytro-ch replied:
Sure, but I would simplify the condition in the following way:
```$storedId = isset($filters['store_id']) ? $filters['store_id'] : null;```

dmytro-ch reviewed (changes requested):
Hi @ivanmiskic,
thank you for your contribution!

Could you please check and fix the issues mentioned in code review comments?

ivanmiskic committed:
Update Placeholder.php

Fixing constructor parameter order for backwards compatibility and checking if parameter exists.

sidolov commented:
@ivanmiskic , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!
